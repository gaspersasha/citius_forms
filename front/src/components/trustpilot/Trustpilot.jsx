import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { generateScriptTag } from '~utils';

const Trustpilot = ({ options }) => {
  const widgetRef = useRef(null);

  useEffect(() => {
    // eslint-disable-next-line
    const { Trustpilot } = window;

    if (Trustpilot) {
      Trustpilot.loadFromElement(widgetRef.current, true);
    } else {
      generateScriptTag(options.scriptPath);
    }
  }, []);

  return (
    <div
      ref={widgetRef}
      className={options.className}
      data-locale={options.locale}
      data-template-id={options.templateId}
      data-businessunit-id={options.businessUnitId}
      data-style-height={options.styleHeigh}
      data-style-width={options.styleWidth}
      data-theme={options.theme}
      data-stars={options.stars}
    >
      <a href={options.accountLink} target="_blank" rel="noopener noreferrer">
        Trustpilot
      </a>
    </div>
  );
};

Trustpilot.propTypes = {
  options: PropTypes.shape({
    scriptPath: PropTypes.string,
    className: PropTypes.string,
    locale: PropTypes.string,
    templateId: PropTypes.string,
    businessUnitId: PropTypes.string,
    styleWidth: PropTypes.string,
    theme: PropTypes.string,
    stars: PropTypes.string,
    styleHeigh: PropTypes.string,
    accountLink: PropTypes.string,
  }).isRequired,
};

export default Trustpilot;
