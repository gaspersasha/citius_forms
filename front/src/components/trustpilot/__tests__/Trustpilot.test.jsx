import React from 'react';

import { generateScriptTag } from '~utils';
import Trustpilot from '../Trustpilot';

jest.mock('~utils', () => ({
  generateScriptTag: jest.fn(),
}));

describe('<Trustpilot />', () => {
  let wrapper;
  const options = {
    scriptPath:
      'https://widget.trustpilot.com/bootstrap/v5/tp.widget.bootstrap.min.js',
    businessUnitId: '4bdcd9ad000064000505fa5a',
    accountLink: 'https://uk.trustpilot.com/review/website',
    locale: 'UA',
    isBigTemplate: false,
    styleHeight: '33%',
    styleWidth: '77%',
    theme: 'light',
    stars: '15',
  };

  beforeEach(() => {
    wrapper = mount(<Trustpilot options={options} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('rendered without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('generateScriptTag invoked', () => {
    expect(generateScriptTag).toBeCalled();

    global.Trustpilot = { loadFromElement: jest.fn() };
  });

  it('trustpilot render method invoked', () => {
    expect(global.Trustpilot.loadFromElement).toBeCalled();
  });
});
