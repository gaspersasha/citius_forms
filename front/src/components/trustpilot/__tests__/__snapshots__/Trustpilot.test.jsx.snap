// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trustpilot /> rendered without errors 1`] = `
<Trustpilot
  options={
    Object {
      "accountLink": "https://uk.trustpilot.com/review/website",
      "businessUnitId": "4bdcd9ad000064000505fa5a",
      "isBigTemplate": false,
      "locale": "UA",
      "scriptPath": "https://widget.trustpilot.com/bootstrap/v5/tp.widget.bootstrap.min.js",
      "stars": "15",
      "styleHeight": "33%",
      "styleWidth": "77%",
      "theme": "light",
    }
  }
>
  <div
    data-businessunit-id="4bdcd9ad000064000505fa5a"
    data-locale="UA"
    data-stars="15"
    data-style-width="77%"
    data-theme="light"
  >
    <a
      href="https://uk.trustpilot.com/review/website"
      rel="noopener noreferrer"
      target="_blank"
    >
      Trustpilot
    </a>
  </div>
</Trustpilot>
`;
